src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -14,497 +14,427 @@ limitations under the License.
-    }
-    if (this.props.selectedNamespace !== ALL_NAMESPACES) {
-      return this.props.selectedNamespace;
-    }
-    return '';
+  initialState(props) {
+    const namespace = props[NAMESPACE];
+    return {
+      ...initialPipelineInfoState(),
+      params: initialParamsState(
+        this.getPipelineInfo(PARAM_SPECS, { state: {}, props })
+      ),
+      resources: initialResourcesState(
+        this.getPipelineInfo(RESOURCE_SPECS, { state: {}, props })
+      ),
+      [NAMESPACE]: namespace !== ALL_NAMESPACES ? namespace : '',
+      serviceAccount: '',
+      timeout: '',
+      validationError: false,
+      submitError: ''
+    };
   }
 
-  checkValidation(key, value) {
-    const { required, regexp } = formValidation[key];
-    const invalid =
-      (required && value === '') ||
-      (regexp && !regexp.test(value) && (required || value !== ''));
-    this.setState(state => {
-      return {
-        [key]: {
-          ...state[key],
-          invalid
-        }
-      };
-    });
-    return !invalid;
+  resetForm() {
+    this.setState((state, props) => this.initialState(props));
   }
 
-  checkFormValidation() {
-    const reducer = (acc, key) => {
-      if (key === 'namespace') {
-        return this.checkValidation(key, this.getNamespace()) && acc;
-      }
-      if (key === 'pipeline') {
-        return this.checkValidation(key, this.getPipeline()) && acc;
-      }
-      return this.checkValidation(key, this.state[key].value) && acc;
-    };
-    return Object.keys(formValidation).reduce(reducer, true);
+  resetParamsState() {
+    this.setState((state, props) => ({
+      params: initialParamsState(
+        this.getPipelineInfo(PARAM_SPECS, { state, props })
+      )
+    }));
   }
 
-  reset() {
-    this.setState(initialState);
+  resetResourcesState() {
+    this.setState((state, props) => ({
+      resources: initialResourcesState(this.getPipelineInfo(RESOURCE_SPECS), {
+        state,
+        props
+      })
+    }));
   }
 
   render() {
-    const { pipelineName, open, namespace } = this.props;
-    const { errorMessage } = this.state;
-    const namespaceValue = this.getNamespace();
-    const pipelineValue = this.getPipeline();
-
+    const { open } = this.props;
+    const { serviceAccount, validationError } = this.state;
+    const namespace = this.getPipelineInfo(NAMESPACE);
+    const pipelineRef = this.getPipelineInfo(PIPELINE_REF);
+    const pipelineInfoDisabled = this.getPipelineInfo(DISABLED);
+    const resourceSpecs = this.getPipelineInfo(RESOURCE_SPECS);
+    const paramSpecs = this.getPipelineInfo(PARAM_SPECS);
     return (
-      <Form onSubmit={this.onSubmit}>
-        <ComposedModal
+      <Form>
+        <Modal
           className="create-pipelinerun"
-          onClose={this.onClose}
           open={open}
+          modalHeading="Create PipelineRun"
+          modalLabel={this.getPipelineInfo(PIPELINE_REF)}
+          primaryButtonText="Submit"
```

Comment:
the primary button text should reflect the action it will perform, in this case 'Create' or 'Create PipelineRun'